### ISSUE TITLE
Add a rule to ensure all props/ state types have all-readonly keys

### ISSUE THREAD

**Issue by lemonmade** [Shopify/eslint-plugin-shopify#202]
More complex. but should be deeply true, so this would be invalid:

```ts
interface Foo {
  id: string;
}

interface State {
  readonly foos: Foo[];
}
```

or maybe even

```ts
interface Foo {
  id: string;
}

interface State {
  readonly foos: ReadonlyArray<Foo>;
}
```
